presbuzz-frontend/src/components/Analytics.js

```diff
@@ -0,0 +1,214 @@
+    if (this.state.redirect === true) {
+      return (
+        <div>
+          <Route path="/articles" render={() => <Articles ids="1,2,3" />} />
+          <Redirect to={"/articles"} />
+        </div>
+      );
+    }
+
+    if (this.state.checked === true) {
+      if (this.state.showComponent === true) {
+        return (
+          <div className="bg" style={style2}>
+            <div className="row">
+              <div className="col-md-12" onClick={this.handleClick}>
+                <ButtonBases className="button" />
+              </div>
+            </div>
+            <div className="container">
+              <div className="row">
+                <div className="col-md-4 box">
+                  <FormGroup row>
+                    <FormControlLabel
+                      control={
+                        <Switch
+                          checked={this.state.checked}
+                          onChange={this.handleChange("checked")}
+                          value="checked"
+                        />
+                      }
+                      label="Compress"
+                    />
+                  </FormGroup>
+                  <Mentions
+                    mentions={Object.keys(this.state.result).length}
+                    size="50px"
+                  />
+                </div>
+                <div className="col-md-4 box">
+                  <h1 className="heading">Year Wise Mentions</h1>
+                  <Barchart data={this.state.result} />
+                </div>
+                <div className="col-md-4 box">
+                  <h1 className="heading">Overall Sentiments</h1>
+                  <Piewheel data={this.state.result} />
+                </div>
+              </div>
+              <div className="row">
+                <div className="col-md-4 box">
+                  <h1 className="heading">Date Wise Trend</h1>
+                  <Linechart data={this.state.result} />
+                </div>
+                <div className="col-md-8 box" style={style}>
+                  <h1 className="heading">In depth view</h1>
+                  <Calendar data={this.state.result} />
+                </div>
+              </div>
+              <div className="row">
+                <div className="col-md-12" style={style}>
+                  <h1 className="heading">Compare Keywords</h1>
+                  <Compare data={this.state.result} />
+                </div>
+              </div>
+            </div>
+            <div className="row">
+              <div className="col-md-12" onClick={this.handleClick}>
+                <ButtonBases />
+              </div>
+            </div>
+          </div>
+        );
+      } else {
+        return (
+          <div>
+            ><h1>Loading...</h1>
+            <CircularProgress disableShrink />
+          </div>
+        );
+      }
+    } else {
+      if (this.state.showComponent === true) {
+        return (
+          <div className="bg" style={style1}>
```

Comment:
In long run, you can create two separate **functional components** in separate files containing the code from these return statements.
Component 1 for checked = false & showComponent=true
Component 2 for checked = true & showComponent=false